examples/Java/HilbertCurve/HilbertCurve.pde

```diff
@@ -0,0 +1,104 @@
+/**
+ * Hilbert Curve
+ * https://www.rosettacode.org/wiki/Hilbert_curve#Processing
+ * Processing 3.4
+ * 2020-03-25 Alexandre Villares
+ * 
+ * Task:
+ * 
+ * Produce a graphical or ASCII-art representation of a Hilbert curve of at least order 3.
+ * https://en.wikipedia.org/wiki/Hilbert_curve
+ */
+
+// L-System based implementation of Hilbert Curve
+
+int  iterations = 7;
+float strokeLen = 600;
+int angleDeg = 90;
+String axiom = "L";
+StringDict rules = new StringDict();
+String sentence = axiom;
+int xo, yo;
+
+void setup() {
+  size(700, 700);
+  xo= 50; 
+  yo = height - 50;
+  strokeWeight(1);
+  noFill();
+  
+  rules.set("L", "+RF-LFL-FR+");
+  rules.set("R", "-LF+RFR+FL-");
+  
+  generate(iterations);
+}
+
+void draw() {
+  background(0);
+  translate(xo, yo);
+  plot(radians(angleDeg));
+}
+
+void generate(int n) {
+  for (int i=0; i < n; i++) {
+    strokeLen *= 0.5;
+    String nextSentence = "";
+    for (int j=0; j < sentence.length(); j++) {
+      char c = sentence.charAt(j);
+      String ruleResult = rules.get(str(c), str(c));
+      nextSentence += ruleResult;
+    }
+    sentence = nextSentence;
+  }
+}
+
+void plot(float angle) {
+  for (int i=0; i < sentence.length(); i++) {
+    char c = sentence.charAt(i);
+    if (c == 'F') {
+      stroke(255); 
+      line(0, 0, 0, -strokeLen);
+      translate(0, -strokeLen);
+    } else if (c == '+') {
+      rotate(angle);
+    } else if (c == '-') {
+      rotate(-angle);
+    } else if (c == '[') {
+      pushMatrix();
+    } else if (c == ']') {
+      popMatrix();
+    }
+  }
+}
+
+void keyPressed() {
+  if (key == '-') {
+    angleDeg -= 1;
+    println("Angle: " + angleDeg);
+  }
+  if (key == '=' || key == '+') {
+    angleDeg += 1;
+    println("Angle: " + angleDeg);
+  }
+  if (key == 's') {
+    saveFrame("####.png");
+  }
```

Comment:
On the wiki I would say this starts to feel like feature creep -- the code is supposed to demonstrate one task, not be a full-featured application. But the navigating and screenshot features are pretty cool, and they don't complicate the rest of the code too much so I figure it is fine to leave them.